scripts/updatebreaches.js

```diff
@@ -0,0 +1,97 @@
+"use strict";
+
+const got = require("got");
+
+const AppConstants = require("../app-constants");
+const HIBP = require("../hibp");
+
+
+const BREACHES_COLLECTION = "fxmonitor-breaches";
+const FX_RS_RECORDS = `${AppConstants.FX_REMOTE_SETTINGS_WRITER_SERVER}/buckets/main-workspace/collections/${BREACHES_COLLECTION}/records`;
+const FX_RS_COLLECTION = `${AppConstants.FX_REMOTE_SETTINGS_WRITER_SERVER}/buckets/main-workspace/collections/${BREACHES_COLLECTION}`;
+const FX_RS_PREVIEW_COLLECTION = `${AppConstants.FX_REMOTE_SETTINGS_WRITER_SERVER}/buckets/main-preview/collections/${BREACHES_COLLECTION}`;
+const FX_RS_USERNAME = AppConstants.FX_REMOTE_SETTINGS_USERNAME;
+const FX_RS_PASSWORD = AppConstants.FX_REMOTE_SETTINGS_PASSWORD;
+
+
+if (!FX_RS_USERNAME || !FX_RS_PASSWORD) {
+  console.error("updatebreaches requires FX_REMOTE_SETTINGS_USERNAME and FX_REMOTE_SETTINGS_PASSWORD.");
+  process.exitCode = 1;
+  return;
+}
+
+
+async function whichBreachesAreNotInRemoteSettingsYet(hibpBreaches) {
+  const remoteSettingsBreachesSet = new Set((await got(
+    `${AppConstants.FX_REMOTE_SETTINGS_ENDPOINT}/buckets/main/collections/${BREACHES_COLLECTION}/records`,
+    { json: true }
+  )).body.data.map(b => b.Name));
+
+  const newBreaches = [];
+  for (const breach of hibpBreaches) {
+    if (remoteSettingsBreachesSet.has(breach.Name)) {
+      continue;
+    }
+    newBreaches.push(breach);
+  }
+  return newBreaches;
+}
+
+
+(async () => {
+  const allHibpBreaches = await HIBP.req("/breaches");
+  const verifiedSiteBreaches = HIBP.filterBreaches(allHibpBreaches);
+
+  const newBreaches = await whichBreachesAreNotInRemoteSettingsYet(verifiedSiteBreaches);
+
+  if (!newBreaches) {
+    console.log("No new breaches found.");
+    return;
+  }
+
+  console.log(`${newBreaches.length} new breach(es) found.`);
+
+  const FX_RS_AUTH = Buffer.from(`${FX_RS_USERNAME}:${FX_RS_PASSWORD}`).toString("base64");
+  const remoteSettingsPreviewSet = new Set((await got(
```

Comment:
Not to self: This should also move into `whichBreachesAreNotInRemoteSettingsYet`